test/e2e/clusterctl_upgrade_test.go

```diff
@@ -34,3 +34,19 @@ var _ = Describe("When testing clusterctl upgrades [clusterctl-Upgrade]", func()
 		}
 	})
 })
+
+var _ = Describe("When testing clusterctl upgrades using ClusterClass", func() {
+	ClusterctlUpgradeSpec(ctx, func() ClusterctlUpgradeSpecInput {
+		return ClusterctlUpgradeSpecInput{
+			E2EConfig:                 e2eConfig,
+			ClusterctlConfigPath:      clusterctlConfigPath,
+			BootstrapClusterProxy:     bootstrapClusterProxy,
+			ArtifactFolder:            artifactFolder,
+			SkipCleanup:               skipCleanup,
+			InitWithBinary:            "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.2.2/clusterctl-{OS}-{ARCH}",
+			InitWithProvidersContract: "v1beta1",
+			InitWithKubernetesVersion: "v1.25.0",
+			WorkloadFlavor:            "topology",
+		}
+	})
+})
```

Comment:
@fabriziopandini I would propose to inline the other jobs we have in test-infra here as well.

I think it's easier to have it in a single place vs. having some tests defined here (as we also have for other test specs) and others in test-infra.